maraboupy/MarabouNetworkNNet.py

```diff
@@ -218,11 +447,204 @@ def nodeTo_f(self, layer, node):
+        Return:
+            (float)
+        """
+        var = self.getVariable(layer, node, f)
+        if self.upperBoundExists(var):
+            return self.upperBounds[var]
+        return None
+
+    def getLowerBound(self, layer, node, f=True):
+        """ Returns lower bound for the variable corresponding to layer, node
+            If f=True, f variable
+            Otherwise, b variable
+        Args:
+            layer (int)
+            node (int)
+            f (bool)
+        Return:
+            (float)
+        """
+        var = self.getVariable(layer, node, f)
+        if self.lowerBoundExists(var):
+            return self.lowerBounds[var]
+        return None
+
+    def getUpperBoundsForLayer(self, layer, f=True):
+        """ Returns a list of upper bounds for the given layer
+            If f=True, f variable
+            Otherwise, b variable
+        Args:
+            layer (int)
+            f (bool)
+        Return:
+            (list of floats)
+        """
+        bound_list = [self.getUpperBound(layer, node, f) for node in range(self.layerSizes[layer])]
+        return bound_list
+
+    def getLowerBoundsForLayer(self, layer, f=True):
+        """ Returns a list of lower bounds for the given layer
+            If f=True, f variable
+            Otherwise, b variable
+        Args:
+            layer (int)
+            f (bool)
+        Return:
+            (list of floats)
+        """
+        bound_list = [self.getLowerBound(layer, node, f) for node in range(self.layerSizes[layer])]
+        return bound_list
+
+    def getBoundsForLayer(self, layer, f=True):
+        """ Returns a tuple of two lists, the lower and upper bounds for the variables corresponding to the given layer
+            If f=True, f variable
+            Otherwise, b variable
+        Args:
+            layer (int)
+            f (bool)
+        Returns:
+            tuple of two list of floats (1st is lower bounds)
+        """
+        return self.getLowerBoundsForLayer(layer, f), self.getUpperBoundsForLayer(layer, f)
+
+
+    def numberOfVariables(self):
+        return self.layerSizes[0] + 2 * sum(self.layerSizes[1:-1]) + 1 * self.layerSizes[-1]
+
+    def getInputMinimum(self, inputVar):
+        return self.inputMinimums[inputVar]
+
+    def getInputMaximum(self, inputVar):
+        return self.inputMaximums[inputVar]
+
+
+    def evaluateWithoutMarabou(self, inputValues):
+        """ Evaluate network directly (without Marabou) at a given point
+        Args:
+            inputValues (list of np arrays): input to network
+        Returns:
+            (np array): output of the network
+        """
+
+        return self.evaluateNNet(inputValues.flatten().tolist(), normalize_inputs=self.normalize,
+                                 normalize_outputs=self.normalize)
+
+
+    def evaluateNNet(self, inputs, first_layer = 0, last_layer=-1, normalize_inputs=False,
+                     normalize_outputs=False, activate_output_layer=False):
```

Comment:
In Args: list, colon should appear after (type). So the arguments should all look like

variable (type): Description